### ISSUE TITLE
Error when parsing function declaration (Throws error when an enum is in function declaration)

### ISSUE THREAD

**Issue by makava** [michaeljones/breathe#223]
Hello, I get the following error when parsing this function:

```
namespace Company { namespace Product {
  template <typename T> class ClassName : private ClassName_Base<T>
  {
    ClassName(T value, enum Company::Product::Communication::ChannelStatus::ECStatus status, OtherProduct::DateTime_t timestamp)
  };
} } // namespaces
```

```
WARNING: Error when parsing function declaration:
If no return type {
Invalid definition: Expected identifier in nested name, got keyword: enum [error at 44]
  Company::Product::ClassName::ClassName(T value, enum Company::Product::Communication::ChannelStatus::ECStatus status, OtherProduct::DateTime_t timestamp)
  ----------------------------------------------------^
} else if return type {
If declId, parameters, and qualifiers {
Invalid definition: Expected identifier in nested name. [error at 30]
  Company::Product::ClassName::ClassName(T value, enum Company::Product::Communication::ChannelStatus::ECStatus status, OtherProduct::DateTime_t timestamp)
  --------------------------------------^
} else If parenthesis in noptr-declarator {
Invalid definition: Expecting "(" in parameters_and_qualifiers. [error at 33]
  Company::Product::ClassName::ClassName(T value, enum Company::Product::Communication::ChannelStatus::ECStatus status, OtherProduct::DateTime_t timestamp)
  -----------------------------------------^
}
```

If I change the function signature in the code to 

`ClassName(T value, OtherProduct::Int32_t status, OtherProduct::DateTime_t timestamp)`
the errors are gone, so it maybe has to do with the enum type?

I have Sphinx Version 1.3.1 and Breathe 4.1.0 installed. 
I also tried with sphinx commit [c9470fa...](https://github.com/sphinx-doc/sphinx/commit/c9470fac65b999dd3289c6c93e5192fc1bb1cca0), where a similar issue was fixed, as well as the latest master branch, but I get the same error with all versions.

Any Ideas why this happens? Is it a Sphinx or a Breathe issue? Thanks for your time and for making this great software!

**Comment by makava** [comment]
Hi again, I investigated a bit and understand now that it is a Sphinx issue, the error arises from the Sphinx funtion `_parse_nested_name(self)` in `domain/cpp.py`. I guess it is not implemented because it's pretty unusual to write the enum keyword in a funtion signature, as we do it (we have that in our coding guidelines), like so:

```ClassName(T value, enum Company::Product::Communication::ChannelStatus::ECStatus status, OtherProduct::DateTime_t timestamp)```

 Anyway, for now I removed the enum keyword from the funtion signature and everything works fine, so I close this issue. However, I'll make a Sphinx issue because it would be nice if this edge case works in the future, because it is valid c++.

Thank you very much for your time, again!
